### PR TITLE
Fix IsInPocketDimension

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -959,7 +959,7 @@ public bool BadgeHidden
         /// <summary>
         /// Gets a value indicating whether or not the player is in the pocket dimension.
         /// </summary>
-        public bool IsInPocketDimension => IsEffectActive<Corroding>() || Map.FindParentRoom(GameObject)?.Type == RoomType.Pocket;
+        public bool IsInPocketDimension => IsEffectActive<Corroding>() && Map.FindParentRoom(GameObject)?.Type == RoomType.Pocket;
```
Comment (Misaka-ZeroTwo) [inline Exiled.API/Features/Player.cs:962]:
```suggestion
        public bool IsInPocketDimension => Map.FindParentRoom(GameObject)?.Type == RoomType.Pocket;
```

I don't see the need to make a check on that effect, considering there isn't much need, and player could still be in pocket dimension without having the effect

Comment (louis1706) [reply Exiled.API/Features/Player.cs:962]: i thinks it's more about getting player getting normally in this room so people without effect like Teleporting player will not be considering in the Room could be good for some plugin to have this

Comment (Misaka-ZeroTwo) [reply Exiled.API/Features/Player.cs:962]: My point was plugin could teleport the player to the pocket dimension and they won't have the effect.

Comment (Thundermaker300) [reply Exiled.API/Features/Player.cs:962]: This should return true if they are physically in the pocket dimension, regardless of whether they have the effect.

**Review by Thundermaker300 — APPROVED** [review]

**Comment by Thundermaker300** [comment]
merge conflict

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -964,7 +964,7 @@ public bool BadgeHidden
         /// <summary>
         /// Gets a value indicating whether or not the player is in the pocket dimension.
         /// </summary>
-        public bool IsInPocketDimension => IsEffectActive<Corroding>() || Room.FindParentRoom(GameObject)?.Type == RoomType.Pocket;
+        public bool IsInPocketDimension => CurrentRoom.Type == RoomType.Pocket;
```
Comment (Thundermaker300) [inline Exiled.API/Features/Player.cs:967]:
```suggestion
        public bool IsInPocketDimension => CurrentRoom?.Type == RoomType.Pocket;
```

Comment (louis1706) [reply Exiled.API/Features/Player.cs:967]: No it's already have null check

**Review by iopietro — APPROVED** [review]

**Pull request merged**